virttest/utils_misc.py

```diff
@@ -2765,20 +2747,56 @@ def get_all_disks_did(session, partition=False):
     :return: a dict with all disks did lists each include disk
              kname, serial and wwn.
     """
-    disks = list_linux_guest_disks(session, partition)
-    logging.debug("Disks detail: %s" % disks)
-    all_disks_did = {}
-    for line in disks:
-        kname = line.split('/')[2]
-        get_disk_info_cmd = "udevadm info -q property -p /sys/block/%s" % kname
-        output = session.cmd_output_safe(get_disk_info_cmd)
-        re_str = r"(?<=DEVNAME=/dev/)(.*)|(?<=ID_SERIAL=)(.*)|"
-        re_str += "(?<=ID_SERIAL_SHORT=)(.*)|(?<=ID_WWN=)(.*)"
-        did_list_group = re.finditer(re_str, output, re.M)
-        did_list = [match.group() for match in did_list_group if match]
-        all_disks_did[kname] = did_list
+    list_disk_cmd = "lsblk -o KNAME,SIZE,TYPE,MOUNTPOINT"
+    output = session.cmd_output_safe(list_disk_cmd)
+    devs = output.splitlines()
+    disks_list = {}
+    mount_list = []
+    for dev in devs:
+        dev = dev.split()
+        if "disk" in dev:
+            disks_list[dev[0]] = dev
+        if partition and "part" in dev:
+            disks_list[dev[0]] = dev
+        if len(dev) == 4:
+            mount_list.append(dev[0])
+            mount_disk = "".join(re.findall(r"([vhs]d[a-z])", dev[0]))
+            if mount_disk not in mount_list:
+                mount_list.append(mount_disk)
+    if not mounted:
+        for mount_disk in mount_list:
+            if mount_disk in disks_list.keys():
+                disks_list.pop(mount_disk)
+    return disks_list
+
+
+def get_windows_disk(session):
+    list_disk_cmd = "wmic volume get driveletter"
+    s, output = session.cmd_status_output(list_disk_cmd)
+    if s != 0:
+        raise exceptions.TestFail("Failed to list disks: %s" % output)
+    disks = re.findall("([A-Z]:)", output)
+    return disks
+
+
+def get_guest_disk(session, ostype, partition=False, mounted=False):
+    if ostype == "windows":
+        return get_windows_disk(session)
+    return get_linux_disk(session, partition, mounted).keys()
 
-    return all_disks_did
+
+def get_linux_disk_did(session, partition=False, mounted=False):
+    """
+    """
+    disks = get_linux_disk(session, partition, mounted)
+    for disk in disks.keys():
+        get_info_cmd = "udevadm info -q property /dev/%s" % disk
```

Comment:
We should use cmd 'udevadm info -q property -p /sys/block/sda' since RHEL6 don't support above one.